core/src/main/java/com/digitalpebble/stormcrawler/protocol/AbstractHttpProtocol.java

```diff
@@ -65,6 +76,134 @@ public void configure(Config conf) {
         robots = new HttpRobotRulesParser(conf);
         protocolMDprefix = ConfUtils.getString(conf,
                 ProtocolResponse.PROTOCOL_MD_PREFIX_PARAM, protocolMDprefix);
+
+        // load proxy rotation from config
+        String proxyRot = ConfUtils.getString(conf, "http.proxy.rotation", "ROUND_ROBIN");
```

Comment:
these configuration details are specific to the MultiProxyManager - they don't need to be known in this class.
Instead of instantiating and configuring the scheme and file, the instance should get the Configuration object and take from there whatever it needs, see for https://github.com/DigitalPebble/storm-crawler/blob/master/core/src/main/java/com/digitalpebble/stormcrawler/filtering/URLFilters.java#L62
This would encapsulate the configuration nicely.